app/src/main/java/org/openobservatory/ooniprobe/activity/RunningActivity.java

```diff
@@ -266,13 +208,58 @@ private void testEnded(Context context) {
     @Override
     public void onConfirmation(Serializable serializable, int i) {
         if (i == DialogInterface.BUTTON_POSITIVE) {
-            if (service != null)
-                service.task.interrupt();
+            if (receiver.service != null)
+                receiver.service.task.interrupt();
```

Comment:
I am not super happy that we're navigating `receiver.service.task.interrupt` and it would perhaps feel a bit more tidy if we could have a method that does the `.service.task.interrupt()` call for us